vernac/declare.ml

```diff
@@ -192,35 +192,35 @@ let { Goptions.get = private_poly_univs } =
     ~value:true
     ()
 
-let universes_of_body_type body typ =
+let universes_of_body_type ~used_univs body typ =
   let used_univs_body = Vars.universes_of_constr body in
-  let used_univs_typ = Option.cata Vars.universes_of_constr Univ.Level.Set.empty typ in
+  let used_univs_typ = Option.cata (Vars.universes_of_constr ~init:used_univs) used_univs typ in
   let used_univs = Univ.Level.Set.union used_univs_body used_univs_typ in
```

Comment:
This could be
```suggestion
  let used_univs = Vars.universes_of_constr body ~init:used_univs_typ in
```
(and remove the `let used_univs_body` above)